### PR TITLE
Fix carousel RTL and refactor code, fix rtl swipe issues

### PR DESCRIPTION
**Pull request by GeoSot**
It fixes rtl next/previous functionality,
it separates directions from order, and transfers all the rtl and direction responsibility on the `_slide` method

Preview: https://deploy-preview-32913--twbs-bootstrap.netlify.app/docs/5.0/components/carousel/

### CODE REVIEW

**Comment by XhmikosR** [comment]
Looks good to me but this reminds me that we don't have tests for this code.

Also, we already have `next()` and `prev()`, perhaps we could leverage them instead of adding new functions?

**Comment by GeoSot** [comment]
I did the  change you propose
Cause of RTL support I would prefer to keep _left/right_ against _next/previous_

**Comment by GeoSot** [comment]
if vertical carousel #32988 is going to be merged, I think the `left/right` is better to be independent of the `next/prev`

**Comment by XhmikosR** [comment]
I'm not sure about #32988; it lacks tests docs and everything. Either way, can you please add a few tests to ensure that when isRTL is true, the right this is done?

**Comment by GeoSot** [comment]
@XhmikosR any help how to mock the isRTL import? I was not able to find any way, so for now I used it as a component method

**Comment by rohit2sharma95** [comment]
@GeoSot You can make `isRTL` a function in the util if you need to check the dir in the runtime :thinking:

**Comment by AqeelAT** [comment]
Hi, I made #33076 before noticing this PR. Would appreciate it if you take a look.

**Comment by rohit2sharma95** [comment]
I think #32446 should be merged first :thinking:

File: js/tests/unit/carousel.spec.js [hunk 1]
```diff
@@ -337,11 +334,11 @@ describe('Carousel', () => {
       const item = fixtureEl.querySelector('#item')
       const carousel = new Carousel(carouselEl)
 
-      spyOn(carousel, 'prev').and.callThrough()
+      spyOn(carousel, '_slide').and.callThrough()
 
       carouselEl.addEventListener('slid.bs.carousel', () => {
         expect(item.classList.contains('active')).toEqual(true)
-        expect(carousel.prev).toHaveBeenCalled()
+        expect(carousel._slide).toHaveBeenCalledWith('right')
```
Comment (rohit2sharma95) [inline js/tests/unit/carousel.spec.js:341]: Can you please also expect that the slide was moved to the right (appropriate) direction? :slightly_smiling_face:

**Comment by GeoSot** [comment]
@rohit2sharma95  I added the checks.
You will find two new methods `directionToOrder` and `orderToDirection`
Both of them, maybe would have to be extracted in future to support other modules with right/left directions

File: js/tests/unit/carousel.spec.js [hunk 2]
```diff
@@ -1061,6 +1062,57 @@ describe('Carousel', () => {
...
+      expect(carousel._directionToOrder('prev')).toEqual('prev')
+      expect(carousel._directionToOrder('right')).toEqual('prev')
+      expect(carousel._directionToOrder('next')).toEqual('next')
+
+      expect(carousel._orderToDirection('next')).toEqual('left')
+      expect(carousel._orderToDirection('prev')).toEqual('right')
+      document.documentElement.dir = 'ltl'
+    })
+
+    it('rtl "_slide" has to call _directionToOrder', () => {
```
Comment (rohit2sharma95) [inline js/tests/unit/carousel.spec.js:1099]: Better if you add check here with rtl as well :thinking:

**Review by rohit2sharma95 — APPROVED** [review]
You may need to update the spec names :slightly_smiling_face:

**Comment by mdo** [comment]
@XhmikosR Will leave for you to squash as you like :)

**Pull request merged**